content/blog/open-sourcing-workshops-on-demand-part3-understanding-the-backend.md

```diff
@@ -0,0 +1,340 @@
+export SYSPRIVDIR=$WODPRIVDIR/sys
+export WODPRIVNOBO=$WODPRIVDIR/notebooks
+WODPRIVINV=""
+# Manages private inventory if any
+if [ -f $WODPRIVDIR/ansible/inventory ]; then
+        WODPRIVINV="-i $WODPRIVDIR/ansible/inventory"
+        export WODPRIVINV
+fi
+
+# AIP-DB PART
+export WODAPIDBDIR=$PWODBEDIR/wod-api-db
+
+# FRONTEND PART
+export WODFEDIR=$PWODBEDIR/wod-frontend
+
+# These dirs are also fixed by default and can be changed as needed
+export WODNOBO=$PWODBEDIR/wod-notebooks
+export STUDDIR=/student
+#
+export ANSPLAYOPT="-e PBKDIR=staging -e WODUSER=wodadmin -e WODBEDIR=/home/wodadmin/wod-backend -e WODNOBO=/home/wodadmin/wod-notebooks -e WODPRIVNOBO=/home/wodadmin/wod-private/notebooks -e WODPRIVDIR=/home/wodadmin/wod-private -e WODAPIDBDIR=/home/wodadmin/wod-api-db -e WODFEDIR=/home/wodadmin/wod-frontend -e STUDDIR=/student -e ANSIBLEDIR=/home/wodadmin/wod-backend/ansible -e ANSIBLEPRIVDIR=/home/wodadmin/wod-private/ansible -e SCRIPTPRIVDIR=/home/wodadmin/wod-private/scripts -e SYSDIR=/home/wodadmin/wod-backend/sys -e SYSPRIVDIR=/home/wodadmin/wod-private/sys"
+export ANSPRIVOPT=" -e @/home/wodadmin/wod-private/ansible/group_vars/all.yml -e @/home/wodadmin/wod-private/ansible/group_vars/staging"
+```
+
+`r﻿andom.sh` ﻿exports the randomly generated password.
+
+`f﻿unctions.sh` is a library of shell functions used by many scripts among which `procmail-action.sh`.  We will see the details shortly below.
+
+4- `procmail-action.sh` c﻿alls the necessary functions and scripts to perform the  **CREATE** operation.
+
+5- `get_session_token()` This function retrieves the necessary token to make api call to the api-db server.
+
+6- `get_workshop_name()`  This function extracts the workshop name from the mail body. In Body: One will find the workshop name: for example, **WKSHP-API101** 
+
+7- `get_workshop_id()` Out of the workshop name, the function `get_workshop_id()` will get the workshop's id from the api-db server.
+
+This id will be used later to get some of the workshop's specifics through additional api calls to the api-db server.
+
+* D﻿oes the workshop require to use the student password as a variable?
+* Does the workshop require ldap authentication?
+* D﻿oes the workshop require a compiled script?
+
+8- `teststdid()` this function checks the student id provided by procmail API is valid: This function exits when the student id is not in the correct range. For each workshop, a dedicated student range is allocated.
+
+9- `generate_randompwd()` This function creates a random password for a user, it is used both for local and ldap users'passwords. If the workshop requires an LDAP authentication (`get_ldap_status()` functions will get this information) then another function is used to update the LDAP server with the password for the given student (`update_ldap_passwd()`)
+
+The generated password will be sent back to the api-db server so that the frontend server can then send an email to allow participant to connect to his workshop.
```

Comment:
API-DB?
Change "his" to "the"
:-) We are trying to be inclusive here... you have a lot of female coders out there now.

Reply:
done